app/models/data_record.rb

```diff
@@ -0,0 +1,24 @@
+class DataRecord < ActiveRecord::Base
```

Comment:
Does this represent a single record from the actual dataset? And it includes some additional information? Are these getter/setter methods overriding ActiveRecord methods so that they can dynamically lookup things?